python-package/README.rst

```diff
@@ -70,17 +70,15 @@ Build from Sources
 
     pip install --no-binary lightgbm lightgbm
 
-For **Linux** and **macOS** users, installation from sources requires installed `CMake`_.
-
-For **Linux** users, **glibc** >= 2.28 is required. Also, in some rare cases you may need to install OpenMP runtime library separately (use your package manager and search for ``lib[g|i]omp`` for doing this).
```

Comment:
Remove this as well?
` For **Linux** users, **glibc** >= 2.14 is required for LightGBM ``<=3.3.3`` and **glibc** >= 2.28 is required for newer versions.`
https://github.com/microsoft/LightGBM/blob/52d40f3ba6952fee82cf63d0ed7f0bf9f279236a/python-package/README.rst?plain=1#L25

Reply:
Yeah I think that makes sense.

At first I'd intentionally left this one in, because it is true that on systems where you have older versions of GLIBC, `pip install lightgbm` will return an sdist (that'll have to be compiled) instead of a wheel.

But looking through it again, I do see that the other sections that explicitly talk about sdists (anything about passing `--no-binary lightgbm`) are nested underneath this "Install from PyPI" heading. And these GLIBC requirements don't affect your ability to build from an sdist.

I've removed this in https://github.com/microsoft/LightGBM/pull/6536/commits/8cee2d623cf54823e0c88de37044f64051ca54e2.